specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/networkData.json

```diff
@@ -0,0 +1,361 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Security Center",
+    "description": "API spec for Microsoft.Security (Azure Security Center) resource provider",
+    "version": "2019-01-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Security/networkData": {
+      "get": {
+        "x-ms-examples": {
+          "subscription List network data": {
+            "$ref": "./examples/networkData/SubscriptionListNetworkData_example.json"
+          }
+        },
+        "tags": [
+          "NetworkData"
+        ],
+        "description": "Get the network data on all your scanned resources inside a scope",
+        "operationId": "NetworkData_List",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/SubscriptionId"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "#/parameters/Expand"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/NetworkDataList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/{resourceId}/providers/Microsoft.Security/NetworkData/default": {
```

Comment:
>NetworkData/default [](start = 48, length = 19)

for singleton patterns you should also be able to perform a collection GET on "/{resourceId}/providers/Microsoft.Security/networkData" and it should return the single instance in a normal collection response syntax

Reply:
In the past we got a comment that this is redundant - we return collection for /subscription/networkData and single object for "/{resourceId}/providers/Microsoft.Security/networkData/default". We saw this is the way on other Rest API's which contains only default object.

Reply:
Its not consistent.  If you do a GET on /networkData and get a 404 but then do a GET on /networkData/default and get a resource that is not RESTful.  Some existing APIs may be doing it wrong but it causes problems downstreams for consumers of your APIs.

Reply:
Agreed.
We will add support for this call as well.

Reply:
fixed.